x-pack/plugins/security_solution/public/detection_engine/rule_management_ui/components/rules_table/add_prebuilt_rules_table/add_prebuilt_rules_flyout.tsx

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React from 'react';
+
+import { useAddPrebuiltRulesTableContext } from './add_prebuilt_rules_table_context';
+import { RuleDetailsFlyout } from '../../../../rule_management/components/rule_details/rule_details_flyout';
+import type { RuleResponse } from '../../../../../../common/api/detection_engine/model/rule_schema/rule_schemas';
+import { diffableRuleToRuleResponse } from '../../../../../../common/detection_engine/diffable_rule_to_rule_response';
+import * as i18n from './translations';
+
+export const AddPrebuiltRulesFlyout = () => {
+  const {
+    state: { flyoutRule, isFlyoutInstallButtonDisabled },
+    actions: { installOneRule, closeFlyout },
+  } = useAddPrebuiltRulesTableContext();
+
+  if (flyoutRule == null) {
+    return null;
+  }
+
+  const ruleResponse: Partial<RuleResponse> = diffableRuleToRuleResponse(flyoutRule);
```

Comment:
Shouldn't this be memoized? The conversion could be expensive in terms of rendering time -- if it's tens of milliseconds it would be perceivable by users.

I know we're going to change the two API endpoints to return `RuleResponse`, so no conversion will be needed, but still wanted to pay attention to this.